aioredis/commands/streams.py

```diff
@@ -0,0 +1,136 @@
+import sys
+from collections import OrderedDict
+
+from aioredis.util import wait_convert
+
+PY_VER = sys.version_info
+
+if PY_VER < (3, 0):
+    from itertools import izip as zip
+
+
+def fields_to_dict(fields):
+    """Convert a flat list of key/values into an OrderedDict"""
+    fields_iterator = iter(fields)
+    return OrderedDict(zip(fields_iterator, fields_iterator))
+
+
+def parse_messages(messages):
+    """ Parse messages as returned by Redis into something useful
+
+    Messages returned by XRANGE arrive in the form:
+
+        [
+            [message_id, [key1, value1, key2, value2, ...]],
+            ...
+        ]
+
+    Here we parse this into:
+
+        [
+            [message_id, OrderedDict(
+                (key1, value1),
+                (key2, value2),
+                ...
+            )],
+            ...
+        ]
+
+    """
+    if messages is None:
+        return []
+    return [(mid, fields_to_dict(values)) for mid, values in messages]
+
+
+def parse_messages_by_stream(messages_by_stream):
+    """ Parse messages returned by stream
+
+    Messages returned by XREAD arrive in the form:
+        [stream_name,
+            [
+                [message_id, [key1, value1, key2, value2, ...]],
+                ...
+            ],
+            ...
+        ]
+
+    Here we parse this into (with the help of the above parse_messages()
+    function):
+
+        [
+            [stream_name, message_id, OrderedDict(
+                (key1, value1),
+                (key2, value2),.
+                ...
+            )],
+            ...
+        ]
+
+    """
+    if messages_by_stream is None:
+        return []
+
+    parsed = []
+    for stream, messages in messages_by_stream:
+        for message_id, fields in parse_messages(messages):
+            parsed.append((stream, message_id, fields))
+    return parsed
+
+
+class StreamCommandsMixin:
+    """Stream commands mixin
+
+    Streams are under development in Redis and
+    not currently released.
+    """
+
+    def xadd(self, stream, fields, message_id=None, max_len=None, exact_len=False):
+        """ Add a message to the specified stream
+        """
+        args = []
+        if max_len is not None:
+            if exact_len:
+                args.extend((b'MAXLEN', max_len))
+            else:
+                args.extend((b'MAXLEN', b'~', max_len))
+
+        args.append(message_id or b'*')
```

Comment:
Why not specifying `message_id=b'*'` in the function arguments ?

Reply:
Good point, done